<launch>
    <node name = "flatness_detect" pkg = "flatness_detect" type = "flatness_detect" output = "screen">
        <param name = "cloud_topic_1" type = "string" value = "/D435i_1/depth/color/points" /><!--/camera/depth/color/points-->
        <param name = "cloud_topic_2" type = "string" value = "/D435i_2/depth/color/points" />
        <param name = "cloud_topic_3" type = "string" value = "/D435i_3/depth/color/points" />
        <param name = "cloud_topic_4" type = "string" value = "/D435i_4/depth/color/points" />
        <param name = "x_GT" type = "double" value = "-0.101" />
        <param name = "x_LT" type = "double" value = "0.151" />
        <param name = "y_GT" type = "double" value = "0.055" />
        <param name = "y_LT" type = "double" value = "0.095" />
        <param name = "z_GT" type = "double" value = "-0.3" />
        <param name = "z_LT" type = "double" value = "0.3" />
        <param name = "ransac_distancethreshold" type = "double" value = "0.0003" />
        <param name = "ransac_maxiterations" type = "int" value = "3000" />
        <param name = "ransac_probability" type = "double" value = "0.9995" />
        <param name = "distance_threshold_" type = "double" value = "0.0055" />
        <param name = "cloud_filtered_width" type = "int" value = "6000" />
        <param name = "kdtree_neighbor_num" type = "int" value = "15" />
        <param name = "kdtree_filter_dis" type = "double" value = "0.0004" />
        <param name = "kdtree_filter_num" type = "int" value = "8" />
        <param name = "using_MBB" type = "bool" value = "false" />
        <param name = "mean_step" type = "int" value = "2" />

        <rosparam command="load" file="$(find flatness_detect)/config/test1.yaml" />
    </node>

    <node name = "flat_rviz" pkg = "rviz" type = "rviz" args="-d $(find flatness_detect)/launch/display.rviz" />


</launch>    
